tools/imas/imas_wrapper.py

```diff
@@ -0,0 +1,542 @@
+	dataEntry.close()
+	
+	ion_names = []
+	ion_densities = []
+	ion_charges = []
+
+	# load data from core profiles with error handling
+	if len(coreprof.profiles_1d[0].grid.rho_tor) == 0:
+		logging.warning('The minor radius from core profiles IDS seems to be empty!\n')
+		logging.error('Core profiles IDS seems to be empty!\n')
+	else:
+		cp_radius = coreprof.profiles_1d[0].grid.rho_tor
+		if log:
+			logging.info('The minor radius from core profiles IDS loaded successfully!\n')
+
+		if len(coreprof.profiles_1d[0].e_field_parallel) != len(cp_radius):
+			logging.warning('The electric field from core profiles IDS seems to be empty!\n')
+			e_field=-999
+		else:
+			e_field = coreprof.profiles_1d[0].e_field_parallel
+			if log:
+				logging.info('The electric field from core profiles IDS loaded successfully!\n')
+
+		if len(coreprof.profiles_1d[0].electrons.temperature) != len(cp_radius):
+			logging.warning('The electron temperature from core profiles IDS seems to be empty!\n')
+			raise Exception('Error in loading core profiles data. Please check the input data!')
+		else:
+			T_cold = coreprof.profiles_1d[0].electrons.temperature
+			if log:
+				logging.info('The electron temperature from core profiles IDS loaded successfully!\n')
+
+		if len(coreprof.profiles_1d[0].electrons.density) != len(cp_radius):
+			logging.warning('The electron density from core profiles IDS seems to be empty!\n')
+			raise Exception('Error in loading core profiles data. Please check the input data!')
+		else:
+			n_e = coreprof.profiles_1d[0].electrons.density
+			if log:
+				logging.info('The electron density from core profiles IDS loaded successfully!\n')
+
+		for i in range(len(coreprof.profiles_1d[0].ion)):
+			
+			if len(coreprof.profiles_1d[0].ion[i].density) != len(cp_radius):
+				logging.warning('An ion density from core profiles IDS seems to be empty!\n')
+				raise Exception('Error in loading core profiles data. Please check the input data!')
+			else:
+				ion_names.append(coreprof.profiles_1d[0].ion[i].label)
+				ion_densities.append(coreprof.profiles_1d[0].ion[i].density)
+				ion_charges.append(int(coreprof.profiles_1d[0].ion[i].z_ion))
+				
+				if log:
+					logging.info('An ion density from core profiles IDS loaded successfully!\n')
+	
+	if coreprof.vacuum_toroidal_field.r0 == -9e+40:
+		logging.warning('The major radius of the magnetic axis from core profiles IDS seems to be empty!\n')
+		raise Exception('Error in loading core profiles data. Please check the input data!')
+	else:
+		R = coreprof.vacuum_toroidal_field.r0
+		if log:
+			logging.info('The major radius of the magnetic axis from core profiles IDS loaded successfully!\n')
+
+	if len(coreprof.vacuum_toroidal_field.b0) == 0:
+		logging.warning('The vacuum toroidal field from core profiles IDS seems to be empty!\n')
+		raise Exception('Error in loading core profiles data. Please check the input data!')
+	else:
+		B0 = coreprof.vacuum_toroidal_field.b0
+		print(B0)
```

Comment:
Maybe this ``print(B0)`` should be removed?

Reply:
I was wondering why I got a number as an output from the example files. I removed it.